var/spack/repos/builtin/packages/py-adb-enhanced/package.py

```diff
@@ -24,4 +25,4 @@ class PyAdbEnhanced(PythonPackage):
     depends_on('py-docopt', type=('build', 'run'))
     depends_on('py-future', type=('build', 'run'))
```

Comment:
Future is also no longer needed. Python dep is now 3.4+.

Reply:
Should we make each dependencies on each version so precisely?
It may be very messy after several version.

Reply:
How about that?
```
    depends_on('python@3:', type=('build', 'run'))
    depends_on('python@3.4:', when='@2.5.10:', type=('build', 'run'))
    depends_on('py-setuptools', type=('build', 'run'))
    depends_on('py-docopt', when='@:2.5.4', type=('build', 'run'))
    depends_on('py-future', when='@:2.5.4', type=('build', 'run'))
    depends_on('py-psutil', type=('build', 'run'))
    depends_on('py-asyncio', when='@:2.5.4', type=('build', 'run'))  
```

Reply:
> Should we make each dependencies on each version so precisely? It may be very messy after several version.

Yes. If you want to reduce the complexity of the package, you could deprecate older versions, but this is standard practice.

> How about that?

Docopt is still needed in the latest version.

Reply:
OK